### PR TITLE
chore: Remove hardDeleteEnrollment [DHIS-17712]

### PR DESCRIPTION
**Pull request by muilpp**
This PR continues the work of moving tracker-related code out of the API module. Specifically, it:
- Removes the hardDeleteEnrollment method from EnrollmentService and EnrollmentStore.
- Removes the ceremony around deleting an event program enrollment.

### CODE REVIEW

**Review by enricocolasante — APPROVED** [review]

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/program/EnrollmentDeletionHandler.java [hunk 1]
```diff
@@ -60,15 +58,10 @@ private DeletionVeto allowDeleteProgram(Program program) {
...
-      while (iterator.hasNext()) {
-        Enrollment enrollment = iterator.next();
-        iterator.remove();
-        enrollmentService.hardDeleteEnrollment(enrollment);
-      }
+    if (program.isRegistration()) {
+      return;
     }
+
+    delete("delete from enrollment where programid = :id", Map.of("id", program.getId()));
```
Comment (teleivo) [inline dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/program/EnrollmentDeletionHandler.java:65]: this is deleting dummy enrollments right?
is it correct that other enrollments into a program with registration are only deleted by trackers TrackerObjectDeletionService?

I probably got all of the above wrong which suggests we should document this at least in a comment here.

**Review by teleivo — APPROVED** [review]

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/program/EnrollmentDeletionHandler.java (hunk 1 above)
Comment (muilpp) [reply dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/program/EnrollmentDeletionHandler.java:65]: you've got it right, but I can add a comment anyway

Comment (teleivo) [reply dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/program/EnrollmentDeletionHandler.java:65]: 😅 yes please

**Review by teleivo — APPROVED** [review]

**Comment by muilpp** [comment]
The api test fails to build the docker image continuously, for some reason, the same PR [here](https://github.com/dhis2/dhis2-core/pull/18176) works, so I close this one

**Pull request closed without merging**